src/btree/bt_curnext.c

```diff
@@ -280,9 +284,22 @@ __cursor_var_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
                 --cbt->recno;
                 continue;
             }
-            WT_RET(__wt_page_cell_data_ref(session, page, &unpack, cbt->tmp));
 
+            cbt->slot = WT_COL_SLOT(page, cip);
+            WT_RET(__wt_txn_read(session, cbt, NULL, &unpack, &upd));
+            if (upd == NULL)
+                continue;
+            if (upd != NULL && upd->type == WT_UPDATE_TOMBSTONE) {
+                if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK))
+                    __wt_free_update_list(session, &upd);
+                continue;
+            }
+
+            WT_RET(__wt_value_return(cbt, upd));
+            cbt->tmp->data = cbt->iface.value.data;
+            cbt->tmp->size = cbt->iface.value.size;
```

Comment:
The same code appears in `__wt_cursor_var_prev` - it's complex and looks error prone - it would be better if you pulled it into a different shared function.

Reply:
The two functions are actually very similar. It is worth to combine them.